environment.go

```diff
@@ -57,9 +120,65 @@ func (e *PipelineEnvironment) exportEnvironment(path string) error {
 
 func (e *PipelineEnvironment) createTemplateParser() *template.Template {
 	fm := template.FuncMap{}
+	// {{ Key }}
+	// Required environment value, if not defined it will not be found as function and raise an error
 	for k, v := range e.Environment {
-		fm[k] = func() string { return *v }
+		// Convert string pointer to static string
```

Comment:
I think this comment is a bit confusing I would just drop it and maybe instead add a comment for why there is a second order function here